docker/daemon.go

```diff
@@ -79,14 +91,37 @@ func migrateKey() (err error) {
 	return nil
 }
 
-func mainDaemon() {
-	if utils.ExperimentalBuild() {
-		logrus.Warn("Running experimental build")
+// DaemonCli represents the daemon CLI.
+type DaemonCli struct {
+	*daemon.Config
+	registryOptions *registry.Options
+}
+
+// CmdDaemon is the daemon command, called the raw arguments after `docker daemon`.
+func (cli *DaemonCli) CmdDaemon(args ...string) error {
+	if *flDaemon {
+		// allow `docker -D -d` and `docker -d -D`
+		logrus.Warn("please use 'docker daemon' instead.")
+	} else if !(commonFlags.FlagSet).IsEmpty() {
+		// deny `docker -D daemon`
+		logrus.Fatal("invalid global flags")
```

Comment:
This error isn't helpful - can we say what's wrong?
I'm seeing this:
```
$ docker --log-level=debug daemon
FATA[0000] invalid global flags
```
but according to the help text it should be legal.